### PR TITLE
(1014) Create a clarification note in the frontend

### PR DESCRIPTION
**Pull request by pezholio**
This allows a user to create a clarification note if there is not enough information to proceed. The copy and design are very rough, but this nails in the journey:

## Screenshots

Assess -- allows me to create and update a clarification note

Assess -- allows me to create and update a clarification note (1)

### CODE REVIEW

**Comment by pezholio** [comment]
[Create a clarification note in the frontend](https://trello.com/c/OPs9wTsg/1014-create-a-clarification-note-in-the-frontend)

File: server/controllers/assess/assessments/clarificationNotesController.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            id: req.params.id,
+            task: 'suitability-assessment',
+            page: 'request-information',
+          }),
+        )
+      }
+    }
+  }
+
+  confirmation(): RequestHandler {
```
Comment (richpjames) [inline server/controllers/assess/assessments/clarificationNotesController.ts:32]: No biggie but we've used 'confirm' over 'confirmation' elsewhere

File: cypress_shared/pages/assess/sufficientInformationPage.ts [hunk 2]
```diff
@@ -5,10 +5,11 @@ import AssessPage from './assessPage'
 import SufficientInformation from '../../../server/form-pages/assess/reviewApplication/sufficientInformation/sufficientInformation'
 
 export default class SufficientInformationPage extends AssessPage {
-  pageClass = new SufficientInformation({ sufficientInformation: 'yes' })
+  pageClass: SufficientInformation
 
-  constructor(assessment: Assessment) {
+  constructor(assessment: Assessment, answer: 'yes' | 'no' = 'yes') {
```
Comment (richpjames) [inline cypress_shared/pages/assess/sufficientInformationPage.ts:10]: We could use the `YesOrNo` type here

File: server/views/assessments/pages/sufficient-information/request-information.njk [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+
+{% from "../../../partials/showErrorSummary.njk" import showErrorSummary %}
+{% from "../../../components/formFields/form-page-textarea/macro.njk" import formPageTextarea %}
+
+{% extends "../../../partials/layout.njk" %}
```
Comment (richpjames) [inline server/views/assessments/pages/sufficient-information/request-information.njk:7]: Can we use the `questions` macro here?

**Review by richpjames — COMMENTED** [review]
Looks good, I've left a couple of comments

File: server/views/assessments/pages/sufficient-information/request-information.njk (hunk 3 above)
Comment (pezholio) [reply server/views/assessments/pages/sufficient-information/request-information.njk:7]: No, because we're POSTing to a different endpoint here 👍

Comment (richpjames) [reply server/views/assessments/pages/sufficient-information/request-information.njk:7]: I assumed there was a reason we couldn't

**Review by richpjames — APPROVED** [review]

**Pull request merged**